js/widgets/forms/clearButton.js

```diff
@@ -20,59 +25,56 @@ define( [
 		_create: function() {
 			this._super();
 
+			if ( !this.inputNeedsWrap ) {
+				return;
+			}
+
 			if ( this.isSearch ) {
 				this.options.clearBtn = true;
 			}
 
-			if ( !!this.options.clearBtn && this.inputNeedsWrap ) {
-				this._addClearBtn();
+			if ( this.options.clearBtn ) {
+				if ( this.options.enhanced ) {
+					this._clearButton = this._outer.children( "a.ui-textinput-clear-button" );
+					this._bindClearEvents();
+				} else {
+					this._addClearButton();
+				}
 			}
 		},
 
 		clearButton: function() {
-			return $( "<a href='#' tabindex='-1' aria-hidden='true' " +
-				"class='ui-input-clear ui-button ui-icon-delete ui-button-icon-only ui-corner-all'>" +
-				"</a>" )
-					.attr( "title", this.options.clearBtnText )
-					.text( this.options.clearBtnText );
-		},
+			var button = $( "<a href='#' tabindex='-1' aria-hidden='true'></a>" )
+				.attr( "title", this.options.clearBtnText )
+				.text( this.options.clearBtnText )
+				.append( "<span>" );
 
-		_clearBtnClick: function( event ) {
-			this.element.val( "" )
-					.focus()
-					.trigger( "change" );
+			this._addClass( button, "ui-textinput-clear-button" );
+			this._addClass( button.children(), "ui-textinput-clear-button-icon" );
```

Comment:
is `children()` correct here seems like you should be separately creating the icon and adding the classes directly to a saved reference?

```js
var icon = $( "<span>" );

this._addClass( button, "ui-textinput-clear-button" );
this._addClass( icon, "ui-textinput-clear-button-icon" );
icon.appendTo( button );
return button;
```